src/main/java/uk/gov/register/core/TransactionalMemoizationStore.java

```diff
@@ -14,30 +15,21 @@ public TransactionalMemoizationStore(MemoizationStore memoizationStore) {
 
     @Override
     public void put(Integer start, Integer size, byte[] value) {
-        if (!stagedHashes.containsKey(start)) {
-            stagedHashes.put(start, new HashMap<>());
-        }
-
-        HashMap<Integer, byte[]> hashesBySize = stagedHashes.get(start);
-        hashesBySize.put(size, value);
+        stagedHashes.put(Pair.of(start, size), value);
     }
 
     @Override
     public byte[] get(Integer start, Integer size) {
-        byte[] hash = null;
-
-        if (stagedHashes.containsKey(start)) {
-            hash = stagedHashes.get(start).get(size);
+        if (stagedHashes.containsKey(Pair.of(start, size)))  {
```

Comment:
maybe assigning `Pair.of(start, size)` to an variable would be clearer and we can reuse it in the `get`?
somink like: `Pair<Integer, Integer> startSizePair = Pair.of(start, size);`  ? what do you think?